src/Microsoft.ServiceFabric.Diagnostics/Metrics/Implementation/Meter.cs

```
@@ -0,0 +1,22 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ServiceFabric.Diagnostics.Metrics.Implementation
+{
+    abstract class Meter
+    {
+        protected readonly IEnumerable<string> systemDimensionValues;
+        protected readonly IFabricMeter fabricMeter;
+
+        internal Meter(IFabricMeter fabricMeter, IEnumerable<string> systemDimensionValues)
+        {
+            this.fabricMeter = fabricMeter ?? throw new ArgumentNullException(nameof(fabricMeter));
+            this.systemDimensionValues = systemDimensionValues ?? new List<string>();
```

Accepting null systemDimensionValues looks odd. I don't see any Meter descendants with constructors that allow null systemDimensionValues. We should fail fast and throw ArgumentNullException unless there is a specific reason for system dimensions to be empty. Even then I think it's better to require a non-null instance that the caller can provide as Enumerable.Empty<String>().

---

I was thinking about a possible future case where we emit metrics outside of a service context. However, you are right, even in that case we can handle this on MeterProvider level and still supply an empty collection.